controllers/object_controls.go

```diff
@@ -3103,6 +3121,18 @@ func transformOpenShiftDriverToolkitContainer(obj *appsv1.DaemonSet, config *gpu
 		}
 	}
 	obj.Spec.Template.Spec.Volumes = append(obj.Spec.Template.Spec.Volumes, volSharedDir)
+
+	// set resource limits
+	if config.Driver.Resources != nil {
+		mainContainer.Resources = corev1.ResourceRequirements{
+			Requests: config.Driver.Resources.Requests,
+			Limits:   config.Driver.Resources.Limits,
+		}
```

Comment:
The function we are currently in, `transformOpenShiftDriverToolkitContainer()`, should only be modifying the driver toolkit container. 
```suggestion
```

Resource requests / limits are applied for the main container in the `transformDriverContainer` function here:
https://github.com/NVIDIA/gpu-operator/blob/d12142c281c74894879b43298e66f4b6171e7100/controllers/object_controls.go#L3289-L3293